src/app/staking/[addressOrENSName]/page.jsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import Tenant from "@/lib/tenant/tenant";
+import { apiFetchStakedDeposits } from "@/app/api/staking/getStakedDeposits";
+import { fetchDelegate } from "@/app/api/common/delegates/getDelegates";
+import { HStack } from "@/components/Layout/Stack";
+import { DepositList } from "@/app/staking/[addressOrENSName]/deposits/DepositList";
+import { StakingStats } from "@/app/staking/components/StakingStats";
+import StakingFaq from "@/app/staking/components/StakingFaq";
+import { PanelClaimRewards } from "@/app/staking/components/PanelClaimRewards";
+import { PanelNewDeposit } from "@/app/staking/components/PanelNewDeposit";
+import { resolveENSName } from "@/app/lib/ENSUtils";
+import { revalidatePath } from "next/cache";
+import { StakingIntro } from "@/app/staking/components/StakingIntro";
+import { PanelSetAllowance } from "@/app/staking/components/PanelSetAllowance";
+
+async function fetchDeposits(address) {
+  "use server";
+  return apiFetchStakedDeposits({ address });
+}
+
+async function apiFetchDelegate(address) {
+  "use server";
+  return fetchDelegate(address);
+}
+
+export default async function Page({ params: { addressOrENSName } }) {
+  const address = (await resolveENSName(addressOrENSName)) || addressOrENSName;
+
+  const { token } = Tenant.current();
+  const { ui, contracts } = Tenant.current();
+  const deposits = await fetchDeposits(address.toLowerCase());
+
+  if (!ui.toggle("staking")) {
+    return <div>Route not supported for namespace</div>;
+  }
+
+  const [totalSupply, totalStaked, rewardPerToken, rewardDuration] =
+    await Promise.all([
+      contracts.token.contract.totalSupply(),
+      contracts.staker.contract.totalStaked(),
```

Comment:
how have you been thinking about revalidating fetches like this? Typically in a non-RSC model I'd fetch this data in a useQuery hook where it can be easily mutated or invalidated but this just seems like a raw fetch -- does it mean we need to hard refresh the page to get it to update if we need an update on something like totalStaked (which could be dynamic if I'm understanding what total staked is)